mantis-network/src/main/java/io/reactivex/mantis/network/push/ObservableTrigger.java

```diff
@@ -155,60 +162,68 @@ private static <K, V> PushTrigger<KeyValuePair<K, V>> groupTrigger(final String
-                        })
-                        .doOnUnsubscribe(() -> {
-                            logger.info("Subscription is INACTIVE for observable trigger with name: " + name);
-                            subscriptionActive.set(0);
-                        })
-                        .flatMap((final GroupedObservable<K, V> group) -> {
-                                    final byte[] keyBytes = keyEncoder.call(group.getKey());
-                                    final long keyBytesHashed = hashFunction.computeHash(keyBytes);
-                                    return
-                                            group
-                                                    .timeout(groupExpirySeconds, TimeUnit.SECONDS, Observable.empty())
-                                                    .lift(new DisableBackPressureOperator<V>())
-                                                    .buffer(250, TimeUnit.MILLISECONDS)
-                                                    .filter((List<V> t1) -> t1 != null && !t1.isEmpty())
-                                                    .map((List<V> list) -> {
-                                                                List<KeyValuePair<K, V>> keyPairList = new ArrayList<>(list.size());
-                                                                for (V data : list) {
-                                                                    keyPairList.add(new KeyValuePair<>(keyBytesHashed, keyBytes, data));
-                                                                }
-                                                                return keyPairList;
-                                                            }
-                                                    );
-                                }
-                        )
-                        .subscribe((List<KeyValuePair<K, V>> list) -> {
-                                    for (KeyValuePair<K, V> data : list) {
-                                        queue.write(data);
-                                    }
-                                },
-                                (Throwable e) -> {
-                                    logger.warn("Observable used to push data errored, on server with name: " + name, e);
-                                    if (doOnError != null) {
-                                        doOnError.call(e);
-                                    }
-                                },
-                                () -> {
-                                    logger.info("Observable used to push data completed, on server with name: " + name);
-                                    if (doOnComplete != null) {
-                                        doOnComplete.call();
-                                    }
-                                }
-
-                        ));
+                    // decouple from calling thread
+                    .observeOn(Schedulers.computation())
```

Comment:
Thoughts on computation scheduler vs IO scheduler for this task?

Reply:
Actually I wonder why we need to specify a schedule at all. It can just run on whatever scheduler it's already running on. @hmitnflx do you know? Maybe I can just remove this line.